### PR TITLE
[16.0] [ADD] stock_release_channel_auto_release

### PR DESCRIPTION
**Pull request by lmignon**
* [ ] includes #506
* [ ] depends on #478 
* [ ] depends on https://github.com/OCA/stock-logistics-warehouse/pull/1532
* [ ] depends on https://github.com/OCA/stock-logistics-warehouse/pull/1560
* [ ] depends on https://github.com/OCA/stock-logistics-warehouse/pull/1681

This addon define an automatic release mode on the release channels. By default
release channels manage the release of the transfers in batch mode. This means
that the transfers are released only when the user manually triggers the release
process by clicking on the release button.

When the automatic release mode is enabled, the transfers are released automatically
when a new transfer is added to the channel or as soon a product becomes available.

As for the batch mode, the automatic release process is only active on open channels.
When is locked, the automatic release process is stopped. Once the channel is unlocked,
the automatic release process is restarted and transfers into the channel are released
if they are ready to be released (IOW if quantities are available for moves not
yet released).

### CODE REVIEW

**Comment by lmignon** [comment]
@jbaudoux @sebalix @simahawk 
Tests are failing on `stock_available_to_promise_release` since tests are ran on a DB with the `sale` addon installed but `sale` is not into the dependencies of `stock_available_to_promise_release`. The dependency on the `sale` addon comes from `stock_release_channel`. That's the only addon that depends on `sale`. I've the feeling that this new dependency could make the tests randomly failing as soon as a product is created into a test and the `sale` addon is not yet loaded into the registry....

File: stock_release_channel_auto_release/readme/DESCRIPTION.rst [hunk 1]
```diff
@@ -0,0 +1,13 @@
+This addons define an automatic release mode on the release channels. By default
```
Comment (TDu) [inline stock_release_channel_auto_release/readme/DESCRIPTION.rst:1]:
```suggestion
This addon defines an automatic release mode on the release channels. By default
```

**Comment by jbaudoux** [comment]
> @jbaudoux @sebalix @simahawk Tests are failing on `stock_available_to_promise_release` since tests are ran on a DB with the `sale` addon installed but `sale` is not into the dependencies of `stock_available_to_promise_release`. The dependency on the `sale` addon comes from `stock_release_channel`. That's the only addon that depends on `sale`. I've the feeling that this new dependency could make the tests randomly failing as soon as a product is created into a test and the `sale` addon is not yet loaded into the registry....

I think the dependency on sales is not necessary anymore with my change done here in the meantime: https://github.com/OCA/wms/commit/e334ddc0bece05fd562714205655c809e24a6caa

**Comment by lmignon** [comment]
> > @jbaudoux @sebalix @simahawk Tests are failing on `stock_available_to_promise_release` since tests are ran on a DB with the `sale` addon installed but `sale` is not into the dependencies of `stock_available_to_promise_release`. The dependency on the `sale` addon comes from `stock_release_channel`. That's the only addon that depends on `sale`. I've the feeling that this new dependency could make the tests randomly failing as soon as a product is created into a test and the `sale` addon is not yet loaded into the registry....
> 
> I think the dependency on sales is not necessary anymore with my change done here in the meantime: [e334ddc](https://github.com/OCA/wms/commit/e334ddc0bece05fd562714205655c809e24a6caa)

@jbaudoux Tests are now :green_circle:

**Comment by sbejaoui** [comment]
@lmignon , @jbaudoux 

https://github.com/acsone/wms/pull/18

File: stock_release_channel/views/stock_release_channel_views.xml [hunk 2]
```diff
@@ -116,6 +191,7 @@
                 class="oe_background_grey o_kanban_dashboard o_emphasize_colors o_stock_release_channel"
                 create="0"
             >
+                <field name="release_mode" invisible="1" />
```
Comment (jbaudoux) [inline stock_release_channel/views/stock_release_channel_views.xml]: I don't think you need the invisible in this section

File: stock_release_channel/views/stock_release_channel_views.xml [hunk 3]
```diff
@@ -148,8 +224,20 @@
                                     <div class="o_kanban_card_header_title">
                                         <div class="o_primary">
                                             <field name="name" />
+                                            <field name="state" invisible="1" />
```
Comment (jbaudoux) [inline stock_release_channel/views/stock_release_channel_views.xml]: I think it should be outside the template in the list of fields to pre-load in the kanban (without invisible)

File: stock_release_channel/views/stock_release_channel_views.xml [hunk 4]
```diff
@@ -166,10 +254,15 @@
                                 <div class="container o_kanban_card_content">
                                     <div class="row o_kanban_card_upper_content">
                                         <div class="col-4 o_kanban_primary_left">
+                                            <field
+                                                name="is_release_allowed"
+                                                invisible="1"
+                                            />
```
Comment (jbaudoux) [inline stock_release_channel/views/stock_release_channel_views.xml]: I think it should be outside the template in the list of fields to pre-load in the kanban (without invisible)

Comment (lmignon) [reply stock_release_channel/views/stock_release_channel_views.xml]: done

File: stock_release_channel/views/stock_release_channel_views.xml (hunk 3 above)
Comment (jbaudoux) [reply stock_release_channel/views/stock_release_channel_views.xml]: You added it in the header but forgot to remove it from here

**Review by jbaudoux — APPROVED** [review]
Thanks for the last cleanup regarding the commercial partner

**Comment by rousseldenis** [comment]
@jbaudoux Could you also review the linked open PR"s? Thanks

**Comment by rousseldenis** [comment]
@TDu Could you update your review ?

**Review by rousseldenis — APPROVED** [review]

**Review by TDu — APPROVED** [review]

**Pull request merged**